tests/test_downscoped.py

```diff
@@ -383,3 +410,256 @@ def test_to_json(self):
                 ]
             }
         }
+
+
+class TestCredentials(object):
+    QUOTA_PROJECT_ID = "QUOTA_PROJECT_ID"
+    GRANT_TYPE = "urn:ietf:params:oauth:grant-type:token-exchange"
+    REQUESTED_TOKEN_TYPE = "urn:ietf:params:oauth:token-type:access_token"
+    TOKEN_EXCHANGE_ENDPOINT = "https://sts.googleapis.com/v1/token"
+    SUBJECT_TOKEN_TYPE = "urn:ietf:params:oauth:token-type:access_token"
+    SUCCESS_RESPONSE = {
+        "access_token": "ACCESS_TOKEN",
+        "issued_token_type": "urn:ietf:params:oauth:token-type:access_token",
+        "token_type": "Bearer",
+        "expires_in": 3600,
+    }
+    ERROR_RESPONSE = {
+        "error": "invalid_grant",
+        "error_description": "Subject token is invalid.",
+        "error_uri": "https://tools.ietf.org/html/rfc6749",
+    }
+    CREDENTIAL_ACCESS_BOUNDARY_JSON = {
+        "accessBoundary": {
+            "accessBoundaryRules": [
+                {
+                    "availablePermissions": AVAILABLE_PERMISSIONS,
+                    "availableResource": AVAILABLE_RESOURCE,
+                    "availabilityCondition": {
+                        "expression": EXPRESSION,
+                        "title": TITLE,
+                        "description": DESCRIPTION,
+                    },
+                }
+            ]
+        }
+    }
+
+    @classmethod
+    def make_credentials(
+        cls, source_credentials=SourceCredentials(), quota_project_id=None
+    ):
+        availability_condition = make_availability_condition(
+            EXPRESSION, TITLE, DESCRIPTION
+        )
+        access_boundary_rule = make_access_boundary_rule(
+            AVAILABLE_RESOURCE, AVAILABLE_PERMISSIONS, availability_condition
+        )
+        rules = [access_boundary_rule]
+        credential_access_boundary = make_credential_access_boundary(rules)
+
+        return downscoped.Credentials(
+            source_credentials, credential_access_boundary, quota_project_id
+        )
+
+    @classmethod
+    def make_mock_request(cls, data, status=http_client.OK):
+        response = mock.create_autospec(transport.Response, instance=True)
+        response.status = status
+        response.data = json.dumps(data).encode("utf-8")
+
+        request = mock.create_autospec(transport.Request)
+        request.return_value = response
+
+        return request
+
+    @classmethod
```

Comment:
And here.

Reply:
I do use `cls` in this method so this may need to remain as a `classmethod`.

Reply:
OK.  Of course, moving the constants to module scope would remove that bit. :)

Reply:
Changed to `staticmethod`.